pixeltable/functions/reve.py

```diff
@@ -0,0 +1,259 @@
+                    return img
+                case 429:
+                    # Try to honor the server-provided Retry-After value if present
+                    # Note: Retry-After value can also be given in the form of HTTP Date, which we don't currently
+                    # handle
+                    retry_after_seconds = None
+                    retry_after_header = resp.headers.get('Retry-After')
+                    if retry_after_header is not None and re.fullmatch(r'\d{1,2}', retry_after_header):
+                        retry_after_seconds = int(retry_after_header)
+                    _logger.info(
+                        f'Reve request {request_id} failed due to rate limiting, retry after header value: '
+                        f'{retry_after_header}'
+                    )
+                    # This error message is formatted specifically so that RequestRateScheduler can extract the retry
+                    # delay from it
+                    raise pxt.Error(
+                        f'Reve request {request_id} failed due to rate limiting (429). retry-after:'
+                        f'{retry_after_seconds}'
+                    )
+                case _:
+                    _logger.error(
+                        f'Reve request {request_id} failed with status code {resp.status} and error code {error_code}'
+                    )
+                    raise pxt.Error(f'Reve request failed with status code {resp.status} and error code {error_code}')
+
+
+@register_client('reve')
+def _(api_key: str) -> _ReveClient:
+    return _ReveClient(api_key=api_key)
+
+
+def _client() -> _ReveClient:
+    return Env.get().get_client('reve')
+
+
+# TODO Regarding rate limiting: Reve appears to be going for a credits per minute rate limiting model, but does not
+# currently communicate rate limit information in responses. Therefore neither of the currently implemented limiting
+# strategies is a perfect match, but "request-rate" is the closest. Reve does not currently enforce the rate limits,
+# but when it does, we can revisit this choice.
+@pxt.udf(resource_pool='request-rate:reve')
+async def create(
+    prompt: str, *, format: str, aspect_ratio: str | None = None, version: str | None = None
+) -> PIL.Image.Image:
+    """
+    Creates an image from a text prompt.
+
+    This UDF wraps the `https://api.reve.com/v1/image/create` endpoint. For more information, refer to the official
+    [API documentation](https://api.reve.com/console/docs/create).
+
+    Args:
+        prompt (str): prompt describing the desired image
+        format (str): desired output image format
+        aspect_ratio (str | None, optional): desired image aspect ratio, e.g. '3:2', '16:9', '1:1', etc.
+        version (str | None, optional): specific model version to use. Latest if not specified.
+
+    Returns:
+        A generated image
+
+    Examples:
+        Add a computed column with generated square images to a table with text prompts:
+        ```
```

Comment:
Use Python code example docstring formatting within `Examples` blocks (see other Pixeltable docstrings e.g. in globals.py for examples)